README.md

```diff
@@ -1 +1,221 @@
-# rust-wasm-template
\ No newline at end of file
+<div align="center">
+  <a href="https://github.com/fission-codes/rust-template" target="_blank">
+    <img src="./assets/logo.png" alt="rust-template Logo" width="100"></img>
+  </a>
+
+  <h1 align="center">rust-template</h1>
+
+  <p>
+    <a href="https://github.com/fission-codes/rust-template/actions?query=">
+      <img src="https://github.com/fission-codes/rust-template/actions/workflows/build.yml/badge.svg" alt="Build Status">
+    </a>
+    <a href="./LICENSE-APACHE">
+      <img src="https://img.shields.io/badge/License-Apache%202.0-blue.svg" alt="License-Apache">
+    </a>
+    <a href="./LICENSE-MIT">
+      <img src="https://img.shields.io/badge/License-MIT-blue.svg" alt="License-MIT">
+    </a>
+    <a href="https://discord.gg/AunfpqKUHU">
+      <img src="https://img.shields.io/static/v1?label=Discord&message=join%20us!&color=mediumslateblue" alt="Discord">
+    </a>
+  </p>
+</div>
+
+<div align="center"><sub>:warning: Work in progress :warning:</sub></div>
+
+##
+
+This is Fission's opinionated [Rust][rust] and [Rust][rust]+[WebAssembly][wasm]
+(wasm) project generator, which leverages the [cargo-generate][cargo-generate]
+tool.
+
+These templates provide various features for getting-up and running with Rust or
+Rust *and* Wasm, including:
+
+* README standardization, code of conduct, contribuing guidelines, and
+  a consistent project layout
+* GitHub issue and pull-request templates
+* Default Rust dependencies (particulary for wasm)
+* Release GitHub Action workflow(s) using the
+  [release-please-action][release-please-action] and the
+  [release-please][release-please] deploy strategy (*optional*)
+  * For wasm libraries, this includes publishing to NPM via
+    [wasm-pack][wasm-pack], reliant on the Cargo version for the wasm package.
+* Test, lint, audit, and code coverage (via [Codecov][codecov]) GitHub Action
+  workflows (*optional*)
+* [Pre-commit][pre-commit] and [rustfmt][rustfmt] opinionated defaults
+* [Dependabot][dependabot] support (*optional*)
+* [Nix flake][nix-flake] support (*optional*)
+* A choice of an [Apache][apache], [MIT][mit], or a dual Apache/MIT license
+
+## Outline
+
+- [Project Templates](#project-templates)
+- [Getting Started](#getting-started)
+- [Contributing](#contributing)
+- [Template References](#template-references)
+- [License](#license)
+
+## Project Templates
+
+This repository contains two sub-templates:
+
+* `rust`: for generating a rust-only library or binary/executable project
+* `rust+wasm` for generating a [cargo workspace][cargo-workspace]
+  with a rust-only crate of the project (library or binary) and another crate
+  for wasm-bindings (library-only), meant for execution in [Node.js][node-js]
+  or running in modern browsers and/or with bundlers like [webpack][webpack].
+
+## Getting Started
```

Comment:
Maybe add a step with instructions for installing `cargo generate` with `cargo install cargo-generate`